File: backend-php/Controllers/CartController.php
<?php

class CartController {
    private CartService $cartService;
    
    public function __construct(CartService $cartService) {
        $this->cartService = $cartService;
    }
    
    public function handleRequest(array $segments, string $method): void {
        // Remover 'api' y 'cart' de los segmentos
        array_shift($segments); // api
        array_shift($segments); // cart
        
        if (empty($segments)) {
            if ($method === 'POST') {
                $this->createCart();
            } else {
                $this->methodNotAllowed();
            }
            return;
        }
        
        $action = $segments[0];
        
        if ($action === 'create' && $method === 'POST') {
            $this->createCart();
            return;
        }
        
        $cartId = $action;
        
        if ($method === 'GET') {
            $this->getCart($cartId);
        } else if ($method === 'POST' && isset($segments[1]) && $segments[1] === 'items') {
            $this->addItemToCart($cartId);
        } else if ($method === 'DELETE') {
            if (isset($segments[1]) && $segments[1] === 'items' && isset($segments[2])) {
                $this->removeItemFromCart($cartId, $segments[2]);
            } else {
                $this->clearCart($cartId);
            }
        } else {
            $this->methodNotAllowed();
        }
    }
    
    private function getCart(string $cartId): void {
        $cart = $this->cartService->getOrCreateCart($cartId);
        $this->jsonResponse($cart->toArray());
    }
    
    private function addItemToCart(string $cartId): void {
        $input = json_decode(file_get_contents('php://input'), true);
        
        if (!isset($input['itemType']) || !isset($input['itemId'])) {
            $this->badRequest('itemType and itemId are required');
            return;
        }
        
        try {
            $cart = $this->cartService->addItemToCart(
                $cartId,
                $input['itemType'],
                (int)$input['itemId']
            );
            $this->jsonResponse($cart->toArray());
        } catch (Exception $e) {
            $this->badRequest($e->getMessage());
        }
    }
    
    private function removeItemFromCart(string $cartId, string $itemId): void {
        $cart = $this->cartService->removeItemFromCart($cartId, $itemId);
        $this->jsonResponse($cart->toArray());
    }
    
    private function clearCart(string $cartId): void {
        $cart = $this->cartService->clearCart($cartId);
        $this->jsonResponse($cart->toArray());
    }
    
    private function createCart(): void {
        $cartId = $this->generateUuid();
        $cart = $this->cartService->getOrCreateCart($cartId);
        $this->jsonResponse($cart->toArray());
    }
    
    private function generateUuid(): string {
        return sprintf('%04x%04x-%04x-%04x-%04x-%04x%04x%04x',
            mt_rand(0, 0xffff), mt_rand(0, 0xffff),
            mt_rand(0, 0xffff),
            mt_rand(0, 0x0fff) | 0x4000,
            mt_rand(0, 0x3fff) | 0x8000,
            mt_rand(0, 0xffff), mt_rand(0, 0xffff), mt_rand(0, 0xffff)
        );
    }
    
    private function jsonResponse($data, int $statusCode = 200): void {
        http_response_code($statusCode);
        echo json_encode($data);
    }
    
    private function badRequest(string $message): void {
        http_response_code(400);
        echo json_encode(['message' => $message]);
    }
    
    private function methodNotAllowed(): void {
        http_response_code(405);
        echo json_encode(['error' => 'Method not allowed']);
    }
}

File: backend/Controllers/CartController.cs
using Microsoft.AspNetCore.Mvc;
using OrtegaTourism.Models;

namespace OrtegaTourism.Controllers
{
    [ApiController]
    [Route("api/[controller]")]
    public class CartController : ControllerBase
    {
        private readonly CartService _cartService;

        public CartController(CartService cartService)
        {
            _cartService = cartService;
        }

        [HttpGet("{cartId}")]
        public ActionResult<Cart> GetCart(string cartId)
        {
            var cart = _cartService.GetOrCreateCart(cartId);
            return Ok(cart);
        }

        [HttpPost("{cartId}/items")]
        public ActionResult<Cart> AddItemToCart(string cartId, [FromBody] AddItemRequest request)
        {
            try
            {
                var cart = _cartService.AddItemToCart(cartId, request.ItemType, request.ItemId);
                return Ok(cart);
            }
            catch (Exception ex)
            {
                return BadRequest(new { message = ex.Message });
            }
        }

        [HttpDelete("{cartId}/items/{itemId}")]
        public ActionResult<Cart> RemoveItemFromCart(string cartId, string itemId)
        {
            var cart = _cartService.RemoveItemFromCart(cartId, itemId);
            return Ok(cart);
        }

        [HttpDelete("{cartId}")]
        public ActionResult<Cart> ClearCart(string cartId)
        {
            var cart = _cartService.ClearCart(cartId);
            return Ok(cart);
        }

        [HttpPost("create")]
        public ActionResult<Cart> CreateCart()
        {
            var cartId = Guid.NewGuid().ToString();
            var cart = _cartService.GetOrCreateCart(cartId);
            return Ok(cart);
        }
    }

    public class AddItemRequest
    {
        public string ItemType { get; set; } = string.Empty;
        public int ItemId { get; set; }
    }
}